Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+enum class StringOffsetFormat {
+    None = 0,
+    Decimal,
+    Octal,
+    Hexadecimal
+};
+
+size_t minimum_string_length = 4;
+bool show_paths = false;
+auto string_offset_format = StringOffsetFormat::None;
+
+static bool should_print_characters(Vector<u8> const& characters)
+{
+    for (u8 ch : characters) {
+        if (is_ascii_printable(ch) && !is_ascii_space(ch))
+            return true;
+    }
+    return false;
+}
+
+static void print_characters(Vector<u8> const& characters, size_t string_offset_position)
+{
+    switch (string_offset_format) {
+    case StringOffsetFormat::Decimal:
+        out("{:>7d} ", string_offset_position);
+        break;
+    case StringOffsetFormat::Octal:
+        out("{:>7o} ", string_offset_position);
+        break;
+    case StringOffsetFormat::Hexadecimal:
+        out("{:>7x} ", string_offset_position);
+        break;
+    default:
+        break;
+    }
+    outln("{:s}", characters.span());
+}
+
+static int process_characters_in_span(Vector<u8>& characters, AK::Span<u8> const& span)
```

Comment:
pass span by value, but const span of u8 should use ReadonlyBytes:

https://github.com/SerenityOS/serenity/blob/aa08c825ecbfd63ce2f3f62e3360ba3aef157ff7/AK/Forward.h#L58

Reply:
Ok, changed to use ReadonlyBytes